Userland/Libraries/LibGfx/ICOLoader.cpp

```diff
@@ -6,16 +6,14 @@
 
 #include <AK/ByteBuffer.h>
 #include <AK/Debug.h>
-#include <AK/DeprecatedMemoryStream.h>
+#include <AK/MemoryStream.h>
 #include <AK/NonnullOwnPtrVector.h>
 #include <AK/Types.h>
 #include <LibGfx/BMPLoader.h>
 #include <LibGfx/ICOLoader.h>
 #include <LibGfx/PNGLoader.h>
 #include <string.h>
 
-namespace Gfx {
-
```

Comment:
`ICONDIR` and `ICONDIRENTRY` should probably not be outside the `Gfx` namespace. It's better to just move the definition of the traits to the bottom of the file, after the namespace declaration has been closed.

Reply:
placing traits at the end of the source file fails to build.

however, i've updated the PR so that ICONDIR and ICONDIRENTRY entries are now in the Gfx namespace. Had to close the namespace between the structs and the rest of the code to define traits, which isn't so awesome, but it works. :/

Reply:
> placing traits at the end of the source file fails to build.

That sounds like something to investigate, it certainly has always worked just fine for me.

Reply:
>  placing traits at the end of the source file fails to build.

This is because the definition of the subsequent classes likely  cause the compiler to look up `AK::Traits<ICONDIR>` and `AK::Traits<ICONDIRENTRY>`.

Putting a specialization of a template after its first use is a hard error.